orchestration/dagster_orchestration/hca_orchestration/solids/validate_egress.py

```diff
@@ -0,0 +1,92 @@
+import os
+from typing import Any
+
+from dagster import configured, solid, InputDefinition, String, DagsterType
+
+from hca_manage.manage import HcaManage, ProblemCount
+
+
+def problem_count_typecheck(_, value: Any) -> bool:
+    return isinstance(value, ProblemCount)
+
+
+DagsterProblemCount: DagsterType = DagsterType(
+    name="DagsterProblemCount",
+    type_check_fn=problem_count_typecheck,
+    description="A simple named tuple to represent the different types of issues "
+                "present from the post process validation.",
+)
+
+
+POST_VALIDATION_SETTINGS_SCHEMA = {
+    "gcp_env": String,
+    "dataset_name": String,
+}
+
+
+@solid(
+    required_resource_keys={"data_repo_client"},
+    config_schema={
+        **POST_VALIDATION_SETTINGS_SCHEMA,
+        "google_project_name": String,
+    }
+)
+def base_post_import_validate(context) -> DagsterProblemCount:
+    """
+    Checks if the target dataset has any rows with duplicate IDs or null file references.
+    """
+    validator = HcaManage(
+        environment=context.solid_config["gcp_env"],
+        project=context.solid_config["google_project_name"],
+        dataset=context.solid_config["dataset_name"],
+        data_repo_client=context.resources.data_repo_client)
```

Comment:
🔧 This is the wrong position for this param, it needs to be first.

Reply:
keyword parameters are position-independent